### PR TITLE
Update to token only build creation endpoint

### LINKED ISSUE #7: Migrate from /repos to /projects
Need to migrate calls and variables away from the legacy `/repos` API endpoint to the modern `/projects` endpoint.

### PR DESCRIPTION
**Pull request by maprules1000**
Update endpoint to use `/api/v1/projects/org slug/project slug/builds/` instead of the long deprecated `/api/v1/repos/org slug/repo slug/builds/`. Closes #7 

Also updates `createBuild` to use the token-only `/api/v1/builds/` endpoint when only a token is provided.

### CODE REVIEW

File: lib/percy/config.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def debug
     end
 
     def repo
-      @repo ||= Percy::Client::Environment.repo
```
Comment (maprules1000) [inline lib/percy/config.rb]: It would be great to have a little historical context on how this works because this feels like a substantial change?

**Comment by maprules1000** [comment]
Have smoke tested this and it passed!

File: lib/percy/config.rb [hunk 2]
```diff
@@ -42,7 +42,7 @@ def debug
     end
 
     def repo
```
Comment (fotinakis) [inline lib/percy/config.rb]: Just like you did elsewhere we should kill this method entirely — any reference to `repo` is super old and we should be ok to remove nowadays (and this method is almost guaranteed to only have been used internally, or set via env vars). It's not even needed to keep this for backwards compatibility really.

File: spec/lib/percy/config_spec.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@
     expect(config.access_token).to be_nil
     expect(config.api_url).to eq(ENV['PERCY_API'])
     expect(config.debug).to eq(false)
-    expect(config.repo).to eq('percy/percy-client')
+    expect(config.repo).to be_nil
```
Comment (fotinakis) [inline spec/lib/percy/config_spec.rb]: Just drop this :)

File: lib/percy/config.rb [hunk 4]
```diff
@@ -7,7 +7,7 @@ class Config
     # @!attribute debug
     #   @return [Boolean] Whether or not to enable debug logging.
     # @!attribute repo
```
Comment (fotinakis) [inline lib/percy/config.rb]: Let's drop this?

**Pull request merged**